### DISCUSSION TITLE
ArgoCD cannot overwrite or set quoted annotations?

### DISCUSSION THREAD

**Discussion by kastl-ars** [argoproj/argo-cd#19254] [Q&A] [answered]
Hi all,

I am struggling to deploy StackRox using ArgoCD. One issue is that the chart sets helm hook annotations on a PVC, that cause trouble during the ArgoCD sync.

But: Those annotations are done like the helm project recommended/required previously, in that they are quoted:

```
metadata:
  annotations:
    "helm.sh/resource-policy": keep
```

The chart offers a way to remove those annotations. To properly overwrite/delete them, one needs to specify them as quoted in the chart's `values.yaml`.

```
---
customize:
  other:
    persistentvolumeclaim/central-db:
      annotations:
        "helm.sh/hook": null
        "helm.sh/hook-delete-policy": null
        "helm.sh/resource-policy": null
```

This works locally using `helm template`, the PVC does not have those annotations (but stil the other annotations not getting removed).

But using the same values.yaml content when deploying with ArgoCD, **the annotations never get removed**.

I only see unquoted annotations when looking at the actual PVC in ArgoCD or using kubectl. So I also tried setting them without quotes in the `values.yaml`, but again, no change, the PVC still has those annotations.

My guess is that ArgoCD drops the quotes somewhere along the way.

The ArgoCD application is created by a ApplicationSet, pointing at a directory in our GitOps repository containing a Chart.yaml and a values.yaml, where the Chart only pulls in the stackrox chart as a dependency. And the values.yaml thus contains a key named after the actual stackrox chart, which then contains the values for this chart.

Anyone else also experiencing this? And maybe having a solution for this?

Kind Regards
Johannes

P.S.: We are aware of an issue with recent helm versions not being able to properly remove values from subcharts by using `null`, but due to this issue our helm version is still on 3.12.3 that is not affected by this.

**Comment by ChristianCiach** [comment]
I cannot help you with your issue, but I believe you are misguided in thinking that the quotes make any kind of difference.

To a yaml parser, it doesn't matter if you quote string literals or not. After parsing the yaml document, the resulting data structure will be exactly the same. Quotes are only necessary to force the parser to interpret the literal as a string where it would otherwise infer a different type.

So, whatever your issue may be, I can almost guarantee that it has nothing to do with the quotes.

**Reply by kastl-ars** [reply, marked as answer]
Thanks for the reply.

> I cannot help you with your issue, but I believe you are misguided in thinking that the quotes make any kind of difference.

I thought so, too, until I saw two sets of annotations in the helm output (locally). Ones with quotes and ones without...

But in the end this seems to be a combination of how our ArgoCD repository is setup (ApplicationProjects pointing to Chart.yaml files that pull in charts as dependencies) and funny quirks in helm (which cannot overwrite values in subcharts/dependencies). We thought the old 1.12.x version of helm that we are using due to this would work properly, but apparently there is more than one issue with helm.
